prefect_census/syncs.py

```diff
@@ -0,0 +1,220 @@
+"""Module containing tasks and flows for interacting with Census syncs."""
+from httpx import HTTPStatusError
+from prefect import flow, task
+from prefect.logging import get_run_logger
+
+from prefect_census.credentials import CensusCredentials
+from prefect_census.runs import (
+    CensusSyncRunCancelled,
+    CensusSyncRunFailed,
+    CensusSyncRunStatus,
+    wait_census_sync_completion,
+)
+from prefect_census.utils import extract_user_message
+
+
+class CensusSyncTriggerFailed(RuntimeError):
+    """Used to indicate sync triggered."""
+
+    pass
+
+
+@task(
+    name="Trigger Census sync run",
+    description="Triggers a Census sync run for the sync " "with the given sync_id.",
+    retries=3,
+    retry_delay_seconds=10,
+)
+async def trigger_census_sync(credentials: CensusCredentials, sync_id: int) -> dict:
+    """
+    A task to trigger a Census sync run.
+
+    Args:
+        credentials: Credentials for authenticating with Census.
+        sync_id: The ID of the sync to trigger.
+
+    Returns:
+        The run data returned from the Census API.
+
+    Examples:
+        Trigger a Census sync run:
+        ```python
+        from prefect import flow
+
+        from prefect_census.credentials import CensusCredentials
+        from prefect_census.syncs import trigger_census_sync
+
+        @flow
+        def trigger_census_sync_flow():
+            credentials = CensusCredentials(api_key="my_api_key")
+            trigger_census_sync(credentials=credentials, sync_id=42)
+
+        trigger_census_sync_flow()
+        ```
+    """  # noqa
+    logger = get_run_logger()
+
+    logger.info(f"Triggering Census sync run for sync with ID {sync_id}")
+    try:
+        async with credentials.get_client() as client:
+            response = await client.trigger_sync_run(sync_id=sync_id)
+    except HTTPStatusError as e:
+        raise CensusSyncTriggerFailed(extract_user_message(e)) from e
+
+    run_data = response.json()["data"]
+
+    if "id" in run_data:
+        logger.info(
+            f"Census sync run successfully triggered for sync with ID {id}. "
+            "You can view the status of this sync run at "
+            f"https://app.getcensus.com/sync/{id}/sync-history"
+        )
+
+    return run_data["sync_run_id"]
+
+
+@task(
+    name="Get Census sync run id",
+    description="Extracts the run ID from a trigger sync run API response",
+)
+def get_run_id(obj: dict) -> int:
```

Comment:
I don't see anywhere that this task is used. Can it be removed?

Reply:
It's used in the example in sync.py at line 107. Should I remove that as well?

Reply:
Yeah I'd say remove it everywhere. There's a task like this in `prefect-dbt` because I was trying to be clever with the linking of nodes in the radar chart, but it doesn't provide a lot of value. It's still in `prefect-dbt` because it would be a breaking change to remove it.